### PR TITLE
Feature/config file

### PR DESCRIPTION
**Pull request by jeremyjams**
* Introduce config file for `local`, `dev` & `prod`
* Connect to `IOracleConsumer.receiveResult(..)` contract interface

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -58,7 +58,7 @@ TARGET_PRIVATE_KEY=
 ## Build native image
 
 ```
-docker image build -f docker/Dockerfile -t generic-oracle-dapp:local .
+docker image build -f docker/Dockerfile -t generic-oracle-dapp:local --build-arg CONFIG_FILE=config.prod.json .
```
Comment (PierreJeanjacquot) [inline Readme.md:61]:
```suggestion
docker image build -f docker/Dockerfile -t generic-oracle-dapp:local --build-arg CONFIG_FILE=config.local.json .
```

File: src/forward/forwardSigner.ts [hunk 2]
```diff
@@ -32,10 +32,10 @@ export async function signForwardRequest(
...
 
   let provider;
-  if (providerUrl) {
-    provider = new ethers.providers.JsonRpcProvider(providerUrl);
-  } else {
+  if (providerUrl == undefined || providerUrl == "") {
     provider = ethers.getDefaultProvider(chainId);
+  } else {
+    provider = new ethers.providers.JsonRpcProvider(providerUrl);
   }
```
Comment (PierreJeanjacquot) [inline src/forward/forwardSigner.ts:39]: can be shortened with `getDefaultProvider` in all  cases

```js
  const provider = ethers.getDefaultProvider(providerUrl || chainId);
```

File: tests/forward/forwardEnvironment.test.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import {
+  getForwarderApiUrl,
+  getOnChainConfig,
+} from "../../src/forward/forwardEnvironment";
+
+describe("Environment", () => {
+  test("should get environment", async () => {
+    expect(getForwarderApiUrl()).toContain("http");
+    expect(getOnChainConfig(5)).not.toBeUndefined();
+    expect(getOnChainConfig(80001)).not.toBeUndefined();
```
Comment (PierreJeanjacquot) [inline tests/forward/forwardEnvironment.test.ts:10]:
```suggestion
    expect(getOnChainConfig(5)).toBeDefined();
    expect(getOnChainConfig(80001)).toBeDefined();
```

**Review by PierreJeanjacquot — APPROVED** [review]
A few minor enhancements are possible (see comments) but it looks good as is :+1:

File: src/forward/forwardSigner.ts (hunk 2 above)
Comment (jeremyjams) [reply src/forward/forwardSigner.ts]: Ok, updated here https://github.com/iExecBlockchainComputing/generic-oracle-dapp/pull/18/commits/a9928d741a154311037d81f94ea149417e3a186d, thanks :+1:

**Pull request merged**